src/main/kotlin/domain/racing/Game.kt

```diff
@@ -2,33 +2,39 @@ package domain.racing
 
 import util.racing.generator.NumberGenerator
 
-class Game(carAmount: Int, private val trial: Int) {
+class Game(names: List<Name>, private val trial: Int) {
     private val cars: Cars
-    val results: MutableList<Map<Car, String>>
+    private val results: MutableList<Map<Car, String>>
 
     init {
-        validateCarAmount(carAmount)
+        validateNames(names)
         validateTrial(trial)
-        this.cars = Cars.of(carAmount)
+        this.cars = Cars.of(names)
         this.results = mutableListOf()
     }
 
-    fun run(numberGenerator: NumberGenerator) {
-        for (it in 1..trial) {
+    fun run(numberGenerator: NumberGenerator): Winners {
+        repeat(trial) {
             cars.move(numberGenerator)
             results.add(cars.toResult())
         }
+
+        return cars.toWinners()
     }
 
-    private fun validateCarAmount(carAmount: Int) {
-        if (carAmount < 1) {
-            throw IllegalArgumentException("차 갯수는 1개 이상이여야 합니다.")
-        }
+    fun getWinnerNames(): List<String> {
+        return cars.toWinners().winners.map { it.getName() }
+    }
+
+    fun getResult(): List<Map<Car, String>> {
+        return results
+    }
+
+    private fun validateNames(names: List<Name>) {
+        require(names.isNotEmpty()) { throw IllegalArgumentException("차 갯수는 1개 이상이여야 합니다.") }
```

Comment:
require block 활용 👍  require 내에서는 따로 throw문을 작성하지 않아도 내부에서 IllegalArgumentException을 throw합니다. [check](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/check.html)와 어떻게 다른지 찾아보는것도 재밌으실거예요.

```suggestion
        require(names.isNotEmpty()) { "차 갯수는 1개 이상이여야 합니다." }
```

Reply:
require는 IllegalArgumentException를 던지고 check는 IllegalStateException을 던지는군요.
각 상황에 맞게 예외를 써야 할때 각 맞춰서 써야겠군요 ㅎㅎ
(알아서 던져주는지는 좀더 찾아볼껄 그랬네요 ㅠㅠ)
감사합니다.